mobile/PlatformShell.tsx

```diff
@@ -6,30 +7,97 @@ import {
   initialWindowMetrics,
 } from 'react-native-safe-area-context'
 
+import {AnalyticsRootProvider} from '~/features/Analytics/context/AnalyticsRootProvider'
+import {routeToEvent} from '~/features/Analytics/events/route-events'
+import {createPosthogClient} from '~/features/Analytics/helpers/createPosthogClient'
+import {useAnalyticsTracking} from '~/features/Analytics/hooks/useAnalyticsTracking'
 import {useScreenCapture} from '~/features/Settings/hooks/useScreenCapture'
 import {RouterContainer} from '~/kernel/navigation/RouterContainer'
+import {
+  initInstallationId,
+  metricsEnabledStorageKeyManager,
+} from '~/kernel/storage/storages'
 import {ModalProvider} from '~/ui/Modal/context/ModalContext'
 
 import {BackgroundTimerProvider} from './src/hooks/BackgroundTimerContext'
 
 export function PlatformShell({children}: React.PropsWithChildren) {
+  const metricsEnabled = metricsEnabledStorageKeyManager.read()
+
   const {init} = useScreenCapture()
-  // Only enable on Android where permission dialogs trigger auto-logout
+
   const isAndroid = Platform.OS === 'android'
+  const platform = isAndroid ? 'Android' : 'IOS'
+  const client = usePosthogClient(metricsEnabled)
 
   React.useEffect(() => {
     init()
   })
 
   return (
     <SafeAreaProvider initialMetrics={initialWindowMetrics}>
-      <RouterContainer>
-        <ModalProvider>
-          <BackgroundTimerProvider active={isAndroid}>
-            <KeyboardProvider statusBarTranslucent>{children}</KeyboardProvider>
-          </BackgroundTimerProvider>
-        </ModalProvider>
-      </RouterContainer>
+      <AnalyticsRootProvider
+        initialEnabled={metricsEnabled}
+        platform={platform}
+        client={client}
+        metricsEnabledStorage={metricsEnabledStorageKeyManager}
+      >
+        <TrackedRouterContainer>
+          <ModalProvider>
+            <BackgroundTimerProvider active={isAndroid}>
+              <KeyboardProvider statusBarTranslucent>
+                {children}
+              </KeyboardProvider>
+            </BackgroundTimerProvider>
+          </ModalProvider>
+        </TrackedRouterContainer>
+      </AnalyticsRootProvider>
     </SafeAreaProvider>
   )
 }
+
+function usePosthogClient(enabled: boolean) {
+  const installationId = React.useMemo(() => initInstallationId(), [])
+
+  const client = React.useMemo(() => {
+    const apiKey = process.env.EXPO_PUBLIC_POSTHOG_KEY
+    const host = process.env.EXPO_PUBLIC_POSTHOG_HOST
+    if (!apiKey || !host) throw new Error('Analytics client is not configured')
+    const sdk = new PostHog(apiKey, {host, disabled: !enabled})
+    return createPosthogClient({sdk})
+  }, [enabled])
+
+  React.useEffect(() => {
+    if (installationId && enabled) client.identify(installationId)
+  }, [client, installationId, enabled])
+
+  return client
+}
+
+function TrackedRouterContainer({children}: React.PropsWithChildren) {
```

Comment:
TrackedRouterContainer - you dont need this.
Just subscribe to router changes here is an example:
https://docs.expo.dev/router/reference/screen-tracking/

You dont need nothing else, just put it in the root layout